app/controllers/IncomeTaxHistoryController.scala

```diff
@@ -49,8 +49,9 @@ class IncomeTaxHistoryController @Inject()(
 )(implicit ec: ExecutionContext, templateRenderer: TemplateRenderer)
     extends TaiBaseController(mcc) {
 
-  private def getIncomeTaxYear(nino: Nino, taxYear: TaxYear)(implicit hc: HeaderCarrier): Future[IncomeTaxYear] = {
-    val futureMaybeTaxCodes = taxAccountService.taxCodeIncomesV2(nino, taxYear).map(_.toOption)
+  def getIncomeTaxYear(nino: Nino, taxYear: TaxYear)(implicit hc: HeaderCarrier): Future[IncomeTaxYear] = {
+    val futureMaybeTaxCodes =
+      taxAccountService.taxCodeIncomesV2(nino, taxYear).map(_.toOption).recover { case _ => None }
```

Comment:
does .recover just recover exceptions? If so what happens when taxCodeIncomesV2 returns successful but no tax code inside. I say this because for my ticket I removed a load of data from the stubs and it only threw an exception for employmentService.employments. I think taxAccountService.taxCodeIncomesV2 is handled gracefully so not sure if the .recover will be invoked?